cmd/lotus-miner/sectors.go

```diff
@@ -146,6 +160,93 @@ var sectorsStatusCmd = &cli.Command{
 			fmt.Printf("Early:\t\t%v\n", status.Early)
 		}
 
+		if cctx.Bool("partition-info") {
+			fullApi, nCloser, err := lcli.GetFullNodeAPI(cctx)
+			if err != nil {
+				return err
+			}
+			defer nCloser()
+
+			maddr, err := getActorAddress(ctx, cctx)
+			if err != nil {
+				return err
+			}
+
+			mact, err := fullApi.StateGetActor(ctx, maddr, types.EmptyTSK)
+			if err != nil {
+				return err
+			}
+
+			tbs := blockstore.NewTieredBstore(blockstore.NewAPIBlockstore(fullApi), blockstore.NewMemory())
+			mas, err := miner.Load(adt.WrapStore(ctx, cbor.NewCborStore(tbs)), mact)
+			if err != nil {
+				return err
+			}
+
+			errFound := errors.New("found")
+			if err := mas.ForEachDeadline(func(dlIdx uint64, dl miner.Deadline) error {
```

Comment:
Future Improvement: what is the current thinking around the lotus miner creating a persistent index from sector id to partition?  It seems like this linear traversal is fairly common and potentially expensive.  I'm sure big miners have this already but it would be nice to have out of the box with lotus.  If we ask around maybe a mining operation would donate their code too.